kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/utils/TopicToTableResolver.java

```diff
@@ -124,36 +141,19 @@ public static void updateTopicToTable(BigQuerySinkConfig config, String topicNam
    * @return A String resulting match of table for topic name.
    */
   private static String getTopicToTableSingleMatch(BigQuerySinkConfig config, String topicName) {
-    String match = null;
-    String previousPattern = null;
-
-    List<Map.Entry<Pattern, String>> patterns = config.getSinglePatterns(
-        BigQuerySinkConfig.TOPICS_TO_TABLES_CONFIG);
-
-    for (Map.Entry<Pattern, String> pattern : patterns) {
-      Matcher patternMatcher = pattern.getKey().matcher(topicName);
-      if (patternMatcher.matches()) {
-        if (match != null) {
-          String secondMatch = pattern.getKey().toString();
-          throw new ConfigException("Value '" + topicName
-              + "' for property '" + BigQuerySinkConfig.TOPICS_CONFIG
-              + "' matches " + BigQuerySinkConfig.TOPICS_TO_TABLES_CONFIG
-              + " regexes for both '" + previousPattern
-              + "' and '" + secondMatch + "'"
-          );
-        }
-        String formatString = pattern.getValue();
-        try {
-          match = patternMatcher.replaceAll(formatString);
-          previousPattern = pattern.getKey().toString();
-        } catch (IndexOutOfBoundsException err) {
-          throw new ConfigException("Format string '" + formatString
-              + "' is invalid in property '" + BigQuerySinkConfig.TOPICS_TO_TABLES_CONFIG
-              + "'", err);
-        }
-      }
-    }
+    return config.getSingleMatch(topicName, BigQuerySinkConfig.TOPICS_CONFIG, BigQuerySinkConfig.TOPICS_TO_TABLES_CONFIG);
```

Comment:
Since this method depends on the BigQueryConfig only, it seemed to be a kind of a _tangled util method_ - a good candidate to become a part of the config.
Also I thought that somebody might need such functionality in the future and would like to reuse the code.
I can move it back to `TopicToTableResolver` if this argument doesn't convince you.

Reply:
Make sense to me. Let's keep it here. 😄